jni/src/faiss_wrapper.cpp

```diff
@@ -624,6 +697,57 @@ jbyteArray knn_jni::faiss_wrapper::TrainIndex(knn_jni::JNIUtilInterface * jniUti
     return ret;
 }
 
+jbyteArray knn_jni::faiss_wrapper::TrainBinaryIndex(knn_jni::JNIUtilInterface * jniUtil, JNIEnv * env, jobject parametersJ,
```

Comment:
We might need to refactor this to reduce code duplication

Reply:
Will have another PR to have the refactor, we can follow on the issue https://github.com/opensearch-project/k-NN/issues/1773